### PR TITLE
Fix/1977

### LINKED ISSUE #1977: Auto-create system databases at startup IF configured for "single node mode"
## Summary

We keep getting requests for CouchDB to "just work" for single-node, standalone installations. Auto-creating all of the system databases on startup for new installs of this is a common request.

## Desired Behaviour

1. Specify somewhere in the config that this is a single-node install.
1. Upon seeing that at startup, if they don't exist, create the system databases automatically, at least `_users`, `_replicator`, if not also `_global_changes`.

## Possible Solution

Add a new config file setting such as `[couchdb] standalone=true`.

## Additional context

c/f https://github.com/apache/couchdb-docker/issues/142

### PR DESCRIPTION
**Pull request by AGrzes**
## Overview

Implementation of suggestion from https://github.com/apache/couchdb/issues/1977 to set up single node automatically when `[couchdb] standalone=true` is set.

I hope I found a proper place to implement it.

## Testing recommendations

I found no fully automatic tests for `setup` module so I updated prepared manual test script and updated `dev/run` script to facilitate testing.

## Related Issues or Pull Requests

Fixes #1977

## Checklist

- [x] Code is written and works correctly
- [x] Changes are covered by tests
- [x] Any new configurable parameters are documented in `rel/overlay/etc/default.ini`
- [ ] A PR for documentation changes has been made in https://github.com/apache/couchdb-documentation

### CODE REVIEW

**Comment by AGrzes** [comment]
It looks like the automated tests are failing at random in areas untouched by this change.
I observer that with subsequent builds failures occurs in different places and sometimes does not occur at all.

**Comment by janl** [comment]
great start, I’ll see this over the finish line

**Comment by AGrzes** [comment]
@janl Do you need anythin from me at this point?

**Comment by janl** [comment]
@AGrzes not yet, I’ll let you know. I’ll take the liberty of doing small amendments directly.

**Comment by janl** [comment]
swapped the name from `standalone` to `single_node` everywhere, to me more consistent with the rest of our naming. Excellent work overall 👍

File: rel/overlay/etc/default.ini [hunk 1]
```diff
@@ -68,6 +68,9 @@ default_engine = couch
 ; inadvertently abusing partitions resulting in hot shards. The default
 ; is 10GiB. A value of 0 or less will disable partition size checks.
 ;max_partition_size = 10737418240
+;
+; Start node in single_node mode so default databases are created immediately.
+;single_node = true
```
Comment (wohali) [inline rel/overlay/etc/default.ini]: @janl this should be `;single_node = false` because the default is false. I'll file a PR

**Pull request merged**